src/puya/ir/builder/blocks.py

```diff
@@ -164,3 +166,30 @@ def loop_continue(self, source_location: SourceLocation) -> None:
             # TODO: this might be a code error or an internal error
             raise InternalError("continue outside of loop", source_location) from ex
         self.goto(target=targets.on_continue, source_location=source_location)
+
+    def make_labelled_block(
+        self, label: str, source_location: SourceLocation, description: str | None
+    ) -> BasicBlock:
+        try:
+            new_block = self._labelled_blocks[label]
+        except KeyError:
+            new_block = self._labelled_blocks[label] = BasicBlock(
+                source_location=source_location, comment=description
+            )
+        self.goto(new_block)
```

Comment:
could add an else block here and set source_location and comment on the new_block to address the issue you mentioned below?

Reply:
oh for some reason I thought ir nodes were immutable